### PR TITLE
[Enhancement] Improve accuracy of broker load progress

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

The previous implementation of load progress used the `load chunk size` divided by the `total file_size`. This calculation is very inaccurate, especially for formats with built-in compression such as parquet. 
Therefore, we consider dividing the actual `read file_size` by the `total file_size`. For parquet, since the actual read file_size cannot be obtained accurately due to the use of a third-party library, it is estimated based on the ratio of the number of rows read to the total number of rows.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/exec/parquet_scanner.cpp [hunk 1]
```diff
@@ -343,13 +343,31 @@ Status ParquetScanner::next_batch() {
...
     while (!_scanner_eof) {
         auto status = _curr_file_reader->next_batch(&_batch);
         if (status.ok() && _batch->num_rows() == 0) {
             continue;
         } else {
+            if (status.ok()) {
+                _next_batch_counter++;
+                _last_file_scan_rows += _batch->num_rows();
+                if (_next_batch_counter % 32 == 0) {
+                    auto incr_bytes = (int64_t)((double)_last_file_scan_rows / _curr_file_reader->total_num_rows() *
```
Comment (chaoyli) [inline be/src/exec/parquet_scanner.cpp:364]: What's the meaning of `_curr_file_reader->total_num_rows() * _last_file_size`?

File: be/src/exec/parquet_scanner.cpp [hunk 2]
```diff
@@ -343,13 +343,31 @@ Status ParquetScanner::next_batch() {
     SCOPED_RAW_TIMER(&_counter->read_batch_ns);
     _batch_start_idx = 0;
     if (_curr_file_reader == nullptr) {
+        if (_last_range_size - _last_file_scan_bytes != 0) {
```
Comment (chaoyli) [inline be/src/exec/parquet_scanner.cpp:346]: Under which circumstances, the `_last_range_size` is not equal `_last_file_scan_bytes` when switch reader?

File: fe/fe-core/src/main/java/com/starrocks/load/loadv2/BrokerLoadJob.java [hunk 3]
```diff
@@ -414,8 +415,16 @@ public void updateProgess(TReportExecStatusParams params) {
         try {
             super.updateProgess(params);
             if (!loadingStatus.getLoadStatistic().getLoadFinish()) {
-                progress = (int) ((double) loadingStatus.getLoadStatistic().totalSourceLoadBytes() /
-                        loadingStatus.getLoadStatistic().totalFileSize() * 100);
+                if (jobType == EtlJobType.BROKER) {
+                    progress = (int) ((double) loadingStatus.getLoadStatistic().sourceScanBytes() /
+                            loadingStatus.getLoadStatistic().totalFileSize() * 100);
+                    LOG.info("progress: {}, sourceScanBytes: {}, totalFileSize: {}",
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/load/loadv2/BrokerLoadJob.java:421]: If we already show it in the terminal, it's nonsense to print in the log

Comment (chaoyli) [reply fe/fe-core/src/main/java/com/starrocks/load/loadv2/BrokerLoadJob.java:421]: If we use the sourceScanBytes / totalFileSize, the number may be larger than 1?

Comment (meegoo) [reply fe/fe-core/src/main/java/com/starrocks/load/loadv2/BrokerLoadJob.java:421]: maybe, progress larger than 100 will reset to 99.

File: be/src/exec/parquet_scanner.cpp (hunk 2 above)
Comment (meegoo) [reply be/src/exec/parquet_scanner.cpp:346]: The range that parquet actually reads and the division on FE may be different

File: be/src/exec/parquet_scanner.cpp (hunk 1 above)
Comment (meegoo) [reply be/src/exec/parquet_scanner.cpp:364]: `_last_file_scan_rows / _curr_file_reader->total_num_rows() * _last_file_size`

File: be/src/exec/parquet_scanner.cpp [hunk 4]
```diff
@@ -330,13 +330,31 @@ Status ParquetScanner::next_batch() {
...
```
Comment (chaoyli) [inline be/src/exec/parquet_scanner.cpp:333]:
```suggestion
        if (_last_range_size != _last_file_scan_bytes) {
```

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-3.0

**Pull request merged**